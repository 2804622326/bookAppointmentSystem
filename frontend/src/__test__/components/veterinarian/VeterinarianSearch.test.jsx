import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import VeterinarianSearch from '../../../components/veterinarian/VeterinarianSearch.jsx';

jest.mock('../../../components/veterinarian/VeterinarianService.js', () => ({
  getAllSpecializations: jest.fn(() => Promise.resolve({ data: ['Surg'] })),
  findAvailableVeterinarians: jest.fn(() => Promise.resolve({ data: [] }))
}));

test('searches veterinarians', async () => {
  const onSearchResult = jest.fn();
  const { asFragment } = render(<VeterinarianSearch onSearchResult={onSearchResult} />);
  await waitFor(() => screen.getByText('Surg'));
  fireEvent.change(screen.getByRole('combobox'), { target: { value: 'Surg' } });
  fireEvent.click(screen.getAllByRole('button', { name: /search/i })[0]);
  await waitFor(() => expect(onSearchResult).toHaveBeenCalled());
  expect(asFragment()).toMatchSnapshot();
});
