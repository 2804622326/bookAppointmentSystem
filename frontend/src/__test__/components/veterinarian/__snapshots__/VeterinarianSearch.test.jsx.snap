// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searches veterinarians 1`] = `
<DocumentFragment>
  <section
    class="stickyFormContainer"
  >
    <h3>
      Find a Veterinarian
    </h3>
    <form
      class=""
    >
      <div>
        <label
          class="form-label"
        >
          Specialization
        </label>
        <select
          class="form-control"
          name="specialization"
        >
          <option
            value=""
          >
            Select Specialization
          </option>
          <option
            value="Surg"
          >
            Surg
          </option>
        </select>
      </div>
      <fieldset>
        <div
          class="mb-3 row"
        >
          <div
            class="col"
          >
            <div
              class="mb-3 mt-3"
            >
              <div
                class="form-check"
              >
                <input
                  class="form-check-input"
                  type="checkbox"
                />
                <label
                  class="form-check-label"
                  title=""
                >
                  Include Date and Time Availabilty
                </label>
              </div>
            </div>
          </div>
        </div>
      </fieldset>
      <div
        class="d-flex justify-content-center mb-4"
      >
        <button
          class="btn btn-outline-primary"
          type="submit"
        >
          Search
        </button>
        <div
          class="mx-2"
        >
          <button
            class="btn btn-outline-info"
            type="button"
          >
            Clear Search
          </button>
        </div>
      </div>
    </form>
    <div />
  </section>
</DocumentFragment>
`;
